xfn-quotable/main.py

```diff
@@ -0,0 +1,84 @@
+import sys
+
+import requests
+import yaml
+
+import fnio
+
+ANNOTATION_KEY_AUTHOR = "quotable.io/author"
+ANNOTATION_KEY_QUOTE = "quotable.io/quote"
+
+
+def get_quote() -> tuple[str, str]:
+    """Get a quote from quotable.io"""
+    rsp = requests.get("https://api.quotable.io/random")
+    rsp.raise_for_status()
+    j = rsp.json()
+    return (j["author"], j["content"])
+
+
+def read_functionio() -> fnio.FunctionIO:
+    """Read the FunctionIO from stdin."""
+    return yaml.load(sys.stdin.read(), yaml.Loader)
```

Comment:
I think I probably want `safe_load` here - sounds like `load` shouldn't be used for untrusted data (which a FnIO may contain).

Reply:
Also, seems like https://pypi.org/project/trycast/ could be handy to validate that the input is actually a FunctionIO.